src/lavinmq/mqtt/broker.cr

```diff
@@ -78,13 +78,16 @@ module LavinMQ
         @clients.delete client_id
       end
 
-      def publish(packet : MQTT::Publish)
+      def create_properties(packet : MQTT::Publish | MQTT::Will) : AMQP::Properties
+        headers = AMQP::Table.new
+        headers["x-mqtt-retain"] = true if packet.retain?
+        headers["x-mqtt-will"] = true if packet.is_a?(MQTT::Will)
```

Comment:
Is this needed?

Reply:
no, we can do 
```suggestion
        def publish(packet : MQTT::Publish | MQTT::Will)
          headers = AMQP::Table.new
          headers["x-mqtt-retain"] = true if packet.retain?
          headers["x-mqtt-will"] = true if packet.is_a?(MQTT::Will)
          properties = AMQP::Properties.new(headers: headers).tap { |props| props.delivery_mode = packet.qos if packet.responds_to?(:qos) }
        
          rk = topicfilter_to_routingkey(packet.topic)
          # TODO: String.new around payload.. should be stored as Bytes
          msg = Message.new("mqtt.default", rk, String.new(packet.payload), properties)
          @exchange.publish(msg, false)
        end
```

Reply:
Oh, I didn't mean the method but the `x-mqtt-will` header!

Reply:
oh, hehe no we don't need that either!